### ISSUE TITLE
Gtk4-media-gstreamer doesn't play any videos and still shows the placeholder.

### ISSUE THREAD

**Issue by gallickgunner** [msys2/MINGW-packages#18293]
I'm trying to develop an application with gtk4 and want to display a video. Found out that the gstreamer media backend was separated into a separate package `gtk4-media-gstreamer`. I installed it but the video still shows a placeholder image and doesn't play. Is this perhaps related to any missing packages?

1) Install `gtk4-media-gstreamer` by doing  `pacman -Syu mingw-w64-ucrt-x86_64-gtk4-media-gstreamer`
2) Run `gtk4-widget-factory`
3) The video in the demo doesn't play

### Expected behavior

I should be seeing a video here like so

[https://upload.wikimedia.org/wikipedia/commons/thumb/2/2d/Gtk4-widget-factory_demos.png/1200px-Gtk4-widget-factory_demos.png](url)

### Actual behavior

The video only shows a placeholder which should be happening when you only install `gtk4` instead of `gtk4-media-gstreamer`
![image](https://github.com/msys2/MINGW-packages/assets/40718428/3c50314c-47e5-4129-a4c6-1fca34aa4c2c)

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-19045

### MINGW environments affected

- [ ] MINGW64
- [ ] MINGW32
- [X] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

**Comment by lazka** [comment]
Does `pacman -S mingw-w64-ucrt-x86_64-gst-plugins-bad` help? We have split that out too because of too many dependencies.

**Comment by gallickgunner** [comment]
> Does `pacman -S mingw-w64-ucrt-x86_64-gst-plugins-bad` help? We have split that out too because of too many dependencies.

According to msys2 site it conflicts with `gstreamer`  but I tried installing it anyways. The video still doesn't play though

**Comment by lazka** [comment]
weird, it works here: 
![image](https://github.com/msys2/MINGW-packages/assets/991986/1fe2b6a7-ed4a-488d-a2a0-f5d67e073530)

I'll have to try with a clean install

**Comment by gallickgunner** [comment]
Hmm this really is weird. Do I have to have some other things installed outside of Msys2? Here is a dump of `gst-inspect-1.0` listing all the plugins available.

[plugins.txt](https://github.com/msys2/MINGW-packages/files/12488573/plugins.txt)

**Comment by lazka** [comment]
> Hmm this really is weird. Do I have to have some other things installed outside of Msys2 for e.g installing gstreamer binaries for windows or some other things like that? I have installed the binaries though.

Note: Since MSYS2 by default clears PATH there shouldn't be any accidental conflicts like this.

I've tried now in a fresh MSYS2 install and it seems you need `pacman -S mingw-w64-ucrt-x86_64-gst-plugins-good`

**Comment by gallickgunner** [comment]
Yes, now it works. So shouldn't this be listed as a dependency of `gtk4-media-gstreamer` ?

**Comment by lazka** [comment]
I think this is just because gtk4-widget-factory happens to include a .webm file which needs the matroska element for demuxing and vpx for decoding, which both live in "good". Not sure...

**Comment by gallickgunner** [comment]
Hmm but I think that sort of information isn't known to a normal developer who just wants common video formats to work. Since we have already split up the gstreamer backend from base gtk4, we shouldn't further split up and just include all the necessary plugins as dependencies of the `gtk4-media-gstreamer` package so if someone wants to work with videos he wouldn't have to go install separate plugins (good, base, bad.. etc).

**Comment by lazka** [comment]
The base/good/bad/ugly/libva split is not really a split by how common they are but depends on the quality of the implementation and the license/patents.

h264, which is likely a very common codec is in "ugly" for example. AAC decoders are in "bad" and "libav".

No objection to adding things to "optional".

**Comment by gallickgunner** [comment]
Alright thanks. I'm glad the problem got solved, now I can move on with the project :)
